### PR TITLE
samples: wifi: Add STA sample

### PR DESCRIPTION
**Pull request by D-Triveni**
This sample allows Nordic chipset to connect & disconnect to an Access point

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

### CODE REVIEW

**Comment by krish2718** [comment]
License check is failing for code that was already existing, so, @carlescufi should we just override this check and fix that in another PR?

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -208,6 +208,8 @@ Wi-Fi samples
 
 * Added :ref:`wifi_scan_sample` sample for Wi-Fi Scan functionality.
 
+* Added :ref:`wifi_station_sample` sample for Wi-Fi Connect functionality.
```
Comment (richabp) [inline doc/nrf/releases/release-notes-changelog.rst:211]:
```suggestion
* Added :ref:`wifi_station_sample` sample that demonstrates how to connect the Wi-Fi station to a specified access point.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -208,6 +208,8 @@ Wi-Fi samples
 
 * Added :ref:`wifi_scan_sample` sample for Wi-Fi Scan functionality.
```
Comment (richabp) [inline doc/nrf/releases/release-notes-changelog.rst:209]:
```suggestion
* Added :ref:`wifi_scan_sample` sample that demonstrates how to scan for the access points.
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: samples/wifi/sta/README.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+   west build -b nrf7002dk_nrf5340_cpuapp
+
+Testing
+=======
+
+|test_sample|
+
+#. |connect_kit|
+#. |connect_terminal|
+
```
[the remark this answers is not in the document]
Comment (D-Triveni) [reply samples/wifi/sta/README.rst:55]: No

File: samples/wifi/sta/Kconfig [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+config STA_KEY_MGMT_WPA2
+	bool "WPA2 Security"
+	help
+	  Enable for WPA2 Security
+
+config STA_KEY_MGMT_WPA2_256
+	bool "WPA2 SHA 256 Security"
+	help
+	  Enable for WPA2-SHA 256 Security
```
Comment (krish2718) [inline samples/wifi/sta/Kconfig:29]:
```suggestion
	  Enable for WPA2-PSK-256 Security
```

File: samples/wifi/sta/Kconfig [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	int "Security Option"
+	default 0 if STA_KEY_MGMT_NONE
+	default 1 if STA_KEY_MGMT_WPA2
+	default 2 if STA_KEY_MGMT_WPA2_256
+	default 3 if STA_KEY_MGMT_WPA3
+	help
+	  For Security option
+
+config STA_SAMPLE_PASSWORD
+	string "PASSWORD"
```
Comment (krish2718) [inline samples/wifi/sta/Kconfig:46]:
```suggestion
	string "Passphrase (WPA2) or password (WPA3)"
```

File: samples/wifi/sta/prj.conf [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+CONFIG_WIFI=y
+CONFIG_WIFI_NRF700X=y
+
+# WPA supplicant
+CONFIG_WPA_SUPP=y
+
+#Below configs need to be modified based on security
```
Comment (krish2718) [inline samples/wifi/sta/prj.conf:12]:
```suggestion
# Below configs need to be modified based on security
```

**Review by krish2718 — APPROVED** [review]
Minor nits.

File: samples/wifi/sta/README.rst [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+The sample supports the following development kit:
+
+.. table-from-sample-yaml::
+
+Overview
+********
+
+This sample can perform Wi-Fi operations "Connect and Disconnect" in the 2.4GHz and 5GHz bands depending on the capabilities of an Access point.
+
+Using this sample, the development kit can connect to the specified Access Point :abbr:`STA (Station)` mode.
```
[the remark this answers is not in the document]
Comment (richabp) [reply samples/wifi/sta/README.rst]: > 

"in" is missing.

**Review by bama-nordic — APPROVED** [review]
LGTM

File: samples/wifi/sta/README.rst [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        RSSI: -55
+      <inf> sta: IP address: 192.168.48.41
+      <inf> sta: Disconnect requested
+      <inf> sta: Disconnection request done (0)
+      Status: successful
+      ==================
+        State: DISCONNECTED
+
+Dependencies
+============
```
Comment (richabp) [inline samples/wifi/sta/README.rst:82]:
```suggestion
Dependencies
************
```

**Review by richabp — CHANGES_REQUESTED** [review]
a minor change.

**Comment by krish2718** [comment]
For license check issue, please wait for #8936 PR to get merged and do the corresponding changes to STA sample.

**Review by richabp — APPROVED** [review]

File: samples/wifi/sta/Kconfig [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+	  Enable for WPA3 Security
+
+config STA_KEY_MGMT_SELECT
+	int "Security Option"
+	default 0 if STA_KEY_MGMT_NONE
+	default 1 if STA_KEY_MGMT_WPA2
+	default 2 if STA_KEY_MGMT_WPA2_256
+	default 3 if STA_KEY_MGMT_WPA3
+	help
+	  For Security option
```
Comment (nordicjm) [inline samples/wifi/sta/Kconfig:43]: Are these options equivalent to a bitmask or is only one usable? If only one is usable, this should be a choice. Additionally STA_KEY_MGMT_SELECT will always be set to the lowest option, if STA_KEY_MGMT_NONE and STA_KEY_MGMT_WPA3 are selected, STA_KEY_MGMT_SELECT will be 0

File: samples/wifi/sta/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(nrf_wifi_sta)
+
+target_include_directories(app PUBLIC ${ZEPHYR_BASE}/subsys/net/ip)
```
Comment (nordicjm) [inline samples/wifi/sta/CMakeLists.txt:12]: Can this not link with the zephyr networking stack in a cleaner way?

File: samples/wifi/sta/README.rst [hunk 11]
```diff
@@ -0,0 +1,88 @@
+.. _wifi_station_sample:
+
+Wi-Fi: Station
+##############
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Station sample demonstrates how to connect the Wi-Fi station to a specified access point.
```
Comment (nordicjm) [inline samples/wifi/sta/README.rst:10]: Would add `using DHCP` or similar, since there seems to be no static IP fallback for this example

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: samples/wifi/sta/CMakeLists.txt (hunk 10 above)
Comment (D-Triveni) [reply samples/wifi/sta/CMakeLists.txt:12]: @nordicjm  Observed that, the other samples are also doing the same as the header file is a private one.

File: samples/wifi/sta/Kconfig (hunk 9 above)
Comment (D-Triveni) [reply samples/wifi/sta/Kconfig]: Resolved. Please do review

**Review by nordicjm — APPROVED** [review]

**Review by bama-nordic — APPROVED** [review]
LGTM

**Pull request merged**